File: backend/src/main/java/com/carrental/backend/reservation/Reservation.java
package com.carrental.backend.reservation;


import com.carrental.backend.user.User;
import com.carrental.backend.vehicle.Vehicle;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Entity
@Table(name="reservations")
@NoArgsConstructor
@Getter
@Setter
public class Reservation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    private User user;

    @ManyToOne
    private Vehicle vehicle;
    private LocalDate startDate;
    private LocalDate endDate;
    private double totalPrice;
    private String customerName;
    private String customerEmail;
    private String customerPhone;
    private String drivingLicenceId;
    private String insuranceType;
    @Column(name = "gps_included")
    private boolean gpsTrackingActive;
    private boolean youngDriver;

    @Column(length = 1000)
    private String returnNotes;

    @Column(name = "pickup_notes", length = 2000)
    private String pickupNotes;

    @Column(name = "damage_notes", length = 2000)
    private String damageNotes;

    @Column(name = "extra_charges", nullable = false)
    private double extraCharges = 0.0;

    @Enumerated(EnumType.STRING)
    private ReservationStatus status;
}

File: frontend/src/types/Reservation.ts
export const ReservationStatus = {
  PENDING_PAYMENT: 'PENDING_PAYMENT',
  CONFIRMED: 'CONFIRMED',
  ACTIVE: 'ACTIVE',
  COMPLETED: 'COMPLETED',
  CANCELLED: 'CANCELLED',
  PAYMENT_FAILED: 'PAYMENT_FAILED',
} as const;

export type ReservationStatusType = typeof ReservationStatus[keyof typeof ReservationStatus];

export type Reservation = {
  id: number;
  user: { id: number; firstName: string; lastName: string; email: string };
  vehicle: { id: number; brand: string; model: string; year: number; type: string; pricePerDay: number; imageUrl?: string };
  startDate: string;
  endDate: string;
  totalPrice: number;
  customerName?: string;
  customerEmail?: string;
  customerPhone?: string;
  drivingLicenceId?: string;
  insuranceType?: 'none' | 'regular' | 'premium';
  gpsIncluded: boolean;
  youngDriver: boolean;
  pickupNotes?: string;
  returnNotes?: string;
  damageNotes?: string;
  extraCharges?: number;
  status: ReservationStatusType;
};

export type ProcessPickupRequest = {
  pickupNotes?: string;
};

export type ProcessReturnRequest = {
  damageNotes?: string;
  extraCharges?: number;
};

export type UpdateReservationDatesRequest = {
  startDate: string;
  endDate: string;
};

export type CreateReservationRequest = {
  userId: number;
  vehicleId: number;
  startDate: string;
  endDate: string;
  customerName: string;
  customerEmail: string;
  customerPhone: string;
  drivingLicenceId: string;
  insuranceType: 'none' | 'regular' | 'premium';
  gpsIncluded: boolean;
  youngDriver: boolean;
};
